### PR TITLE
Added sleep_sort [C++]

### LINKED ISSUE #346: Sleep Sort [C++]
#### This issue is about [Sleep Sort](http://www.geeksforgeeks.org/sleep-sort-king-laziness-sorting-sleeping/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #346

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: sleep_sort/sleep_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	}
+	for (auto& th : threads) {
+		th.join();
+	}
+}
+
+/*
+ * Driver function
+ * For compilation of the code use -pthread
+ * compilation looks like : g++ sleep_sort.cpp -o sleep_sort -pthread
```
Comment (mohitkyadav) [inline sleep_sort/sleep_sort.cpp]: Please also specify `-std=c++11` flag.

**Comment by yashLadha** [comment]
It works with that too, that i mentioned(if you are using g++).

On Fri, Jun 23, 2017 at 6:59 PM, Mohit Kumar Yadav <notifications@github.com

**Comment by Prakash2403** [comment]
Fix travis-ci error.

**Comment by yashLadha** [comment]
In the travis there is no passing of thread flag during compilation. It has to be done in  a separate Issue @Prakash2403

File: sleep_sort/sleep_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#include <iostream>
```
Comment (Prakash2403) [inline sleep_sort/sleep_sort.cpp]: Mention complexities

Comment (yashLadha) [inline sleep_sort/sleep_sort.cpp]: Added 👍

File: sleep_sort/sleep_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+/*
+ * Driver function
+ * For compilation of the code use -pthread
+ * compilation looks like : g++ -std=c++11 sleep_sort.cpp -o sleep_sort -pthread
+ */
+int main() {
+	int ar[] = {8, 4, 9, 1};
+	vector<int> sorted_number;
```
Comment (aviaryan) [inline sleep_sort/sleep_sort.cpp]: `sorted_numbers` or `sorted` would be a better name. `sorted_number` is misleading.

**Pull request merged**